### PR TITLE
Feat/bac 196 upload questions with csv

### PR DESCRIPTION
**Pull request by henry-mbamalu**
#### Task reference
https://linear.app/team-axle-skript-project/issue/BAC-196/upload-questions-with-csv
https://linear.app/team-axle-skript-project/issue/BAC-185/build-an-endpoint-to-get-question-by-assessment-id
https://linear.app/team-axle-skript-project/issue/BAC-186/build-an-endpoint-to-get-question-by-category-id
https://linear.app/team-axle-skript-project/issue/BAC-184/build-an-endpoint-to-create-question
#### Summary
built an endpoint to upload questions with csv or other excel file
updated the get question by assessment id endpoint
updated the get question by category id endpoint
updated the add question manually endpoint
#### References 

uploadcsv
getassess

### CODE REVIEW

File: backend/database/migrations/2022_11_16_134040_create_questions_table.php [hunk 1]
```diff
@@ -15,13 +15,14 @@ public function up()
     {
         Schema::create('questions', function (Blueprint $table) {
             $table->uuid('id')->unique();
-            $table->string("question");
             $table->string("options");
             $table->string("timeframe");
             $table->string("correct_answers");
+            $table->string("question");
             $table->boolean("is_multiple_answers");
```
Comment (dasaint001) [inline backend/database/migrations/2022_11_16_134040_create_questions_table.php:22]: this is not accepted.. pls use the update migration method instead..

File: backend/routes/api.php [hunk 2]
```diff
@@ -80,7 +80,7 @@
 
 //Assessment routes operations
 Route::prefix("assessment")->group(function () {
-    Route::post('/create', [AssessmentController::class, 'createAssessment'])->middleware("isloggedin", "isadmin");
+    Route::post('/create', [AssessmentController::class, 'createAssessment']);
```
Comment (dasaint001) [inline backend/routes/api.php]: I guess you forgot to return the middleware here

File: backend/routes/api.php [hunk 3]
```diff
@@ -146,6 +146,7 @@ function () {
 
 // questions route operations
 Route::prefix("question")->group(function () {
+    Route::post('upload', [QuestionsController::class, 'uploadCsv']);
```
Comment (dasaint001) [inline backend/routes/api.php]: why is your middleware missing here too?

File: frontend/src/ui/components/CreateAssessment/SelectType.jsx [hunk 4]
```diff
@@ -1,6 +1,9 @@
 import React, { useEffect, useState } from "react";
 import { Link, useNavigate } from "react-router-dom";
 import styled from "styled-components";
+import axios from "../../../api/axios";
+
+
```
Comment (dasaint001) [inline frontend/src/ui/components/CreateAssessment/SelectType.jsx]: why is this here? this belongs to the frontend section..

**Review by dasaint001 — CHANGES_REQUESTED** [review]

File: backend/database/migrations/2022_11_16_134040_create_questions_table.php (hunk 1 above)
Comment (dicodedev) [reply backend/database/migrations/2022_11_16_134040_create_questions_table.php:22]: Why did you touch the migration file in the first place?

File: frontend/src/ui/components/CreateAssessment/SelectType.jsx (hunk 4 above)
Comment (henry-mbamalu) [reply frontend/src/ui/components/CreateAssessment/SelectType.jsx]: it was something I was working on before, I forgot to remove it

File: backend/database/migrations/2022_11_16_134040_create_questions_table.php (hunk 1 above)
Comment (henry-mbamalu) [reply backend/database/migrations/2022_11_16_134040_create_questions_table.php:22]: i'll remove it, i think it the update migration for this already exists

Comment (dasaint001) [reply backend/database/migrations/2022_11_16_134040_create_questions_table.php:22]: you didn't return the question_id column..

File: backend/database/migrations/2022_11_16_134040_create_questions_table.php [hunk 5]
```diff
@@ -15,13 +15,13 @@ public function up()
     {
         Schema::create('questions', function (Blueprint $table) {
             $table->uuid('id')->unique();
-            $table->string("question");
```
Comment (dasaint001) [inline backend/database/migrations/2022_11_16_134040_create_questions_table.php]: revert this back

**Review by dasaint001 — CHANGES_REQUESTED** [review]

File: backend/database/migrations/2022_11_16_134040_create_questions_table.php (hunk 1 above)
Comment (henry-mbamalu) [reply backend/database/migrations/2022_11_16_134040_create_questions_table.php:22]: it's no more in the dev branch

**Comment by henry-mbamalu** [comment]
createques

File: backend/database/migrations/2022_11_16_134040_create_questions_table.php (hunk 1 above)
Comment (dasaint001) [reply backend/database/migrations/2022_11_16_134040_create_questions_table.php:22]: ok

File: backend/database/migrations/2022_11_16_134040_create_questions_table.php [hunk 6]
```diff
@@ -21,7 +21,8 @@ public function up()
             $table->string("correct_answers");
             $table->boolean("is_multiple_answers");
             $table->string('category_id');
-            $table->string('assessment_id');
+            $table->string('assessment_id'); 
+            $table->string('company_id');
```
Comment (dasaint001) [inline backend/database/migrations/2022_11_16_134040_create_questions_table.php]: company_id is a new introduction. It should also be done using update migration method

File: frontend/src/ui/components/Company/Assessments/CreateAssessment/SelectType.jsx [hunk 7]
```diff
@@ -78,6 +78,8 @@ const SelectType = () => {
   }, [formData, touched]);
```
Comment (dasaint001) [inline frontend/src/ui/components/Company/Assessments/CreateAssessment/SelectType.jsx:79]: this is not forme to merge tho.... pls remove... this belongs to FE

File: frontend/src/api/axios.js [hunk 8]
```diff
@@ -2,7 +2,7 @@ import axios from "axios";
 
 const BASEURL = "https://api.eval360.hng.tech/api/";
 // const BASEURL = "http://104.225.216.199:8000/api/";
-//const BASEURL = "http://localhost:8000/api/";
+// const BASEURL = "http://localhost:8000/api/";
```
Comment (dasaint001) [inline frontend/src/api/axios.js:6]: same here

**Review by dasaint001 — CHANGES_REQUESTED** [review]

File: frontend/src/ui/components/Company/Assessments/CreateAssessment/SelectType.jsx (hunk 7 above)
Comment (tobiezele) [reply frontend/src/ui/components/Company/Assessments/CreateAssessment/SelectType.jsx:79]: @dasaint001 Invite the FE lead to look at the aspect he is to merge.

File: frontend/src/api/axios.js (hunk 8 above)
Comment (henry-mbamalu) [reply frontend/src/api/axios.js:6]: nothing changed here

Comment (henry-mbamalu) [reply frontend/src/api/axios.js:6]: it's still commented out

File: frontend/src/ui/components/Company/Assessments/CreateAssessment/SelectType.jsx (hunk 7 above)
Comment (henry-mbamalu) [reply frontend/src/ui/components/Company/Assessments/CreateAssessment/SelectType.jsx:79]: it's someone from frontend that added it, if I remove it it'll cause conflict, it was added earlier when i was resolving conflict, part of the incoming change

Comment (dasaint001) [reply frontend/src/ui/components/Company/Assessments/CreateAssessment/SelectType.jsx:79]: and merging this can also cause conflicts in the FE folder...

Comment (dasaint001) [reply frontend/src/ui/components/Company/Assessments/CreateAssessment/SelectType.jsx:79]: Please if you have anything to do in the frontend folder, create a separate PR for that...

**Review by dasaint001 — APPROVED** [review]
